### PR TITLE
Crd definition

### PR DESCRIPTION
**Pull request by xamebax**
Adds the following:
- Custom Resource Definition
- autogenerated code
- `hack` folder with update/verify scripts

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# FIAAS Custom Resource Definition
+
+This library contains a Custom Resource Definition and an auto-generated Go client used to manipulating the Application FIAAS resource.
+
+Code responsible for generating this CRD is present in `pkg/apis/fiaas.schibsted.io/v1/`:
+
+- `types.go` was modeled after https://github.schibsted.io/finn/fiaas-deploy-daemon/blob/master/fiaas_deploy_daemon/crd/types.py
```
Comment (oyvindio) [inline README.md]: This should probably point to github.com/fiaas/fiaas-deploy-daemon

File: README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# FIAAS Custom Resource Definition
+
+This library contains a Custom Resource Definition and an auto-generated Go client used to manipulating the Application FIAAS resource.
```
Comment (oyvindio) [inline README.md]: "to manipulating" should maybe be "for manipulating" or "to manipulate"

**Review by oyvindio — APPROVED** [review]
I added some minor nitpicks inline, otherwise this looks good

File: hack/boilerplate.go.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/*
+Copyright The Kubernetes Authors.
```
Comment (mortenlj) [inline hack/boilerplate.go.txt:2]: This is probably not the right copyright?

File: hack/update-codegen.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TEMP_DIR=`mktemp -d`
+SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
+CODEGEN_PACKAGE=$TEMP_DIR/code-generator
+
+git clone -b kubernetes-1.13.1 https://github.com/kubernetes/code-generator $CODEGEN_PACKAGE
```
Comment (mortenlj) [inline hack/update-codegen.sh:11]: What is `kubernetes-1.13.1` here? Is it the kubernetes version the generated code targets? Currently we are (for reasons) targetting 1.9 and up. Would this still work in 1.9? If not, it should probably be documented in the README that this library targets a specific kubernetes version that is more strict than FIAAS in general.

File: pkg/apis/fiaas.schibsted.io/v1/doc.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```
Comment (mortenlj) [inline pkg/apis/fiaas.schibsted.io/v1/doc.go:2]: This copyright does not look right.

Comment (mortenlj) [reply pkg/apis/fiaas.schibsted.io/v1/doc.go:2]: I see the same problem in several files. We should probably check that we have the right copyright everywhere.

File: pkg/client/clientset/versioned/clientset.go [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+	*discovery.DiscoveryClient
+	fiaasV1 *fiaasv1.FiaasV1Client
+}
+
+// FiaasV1 retrieves the FiaasV1Client
+func (c *Clientset) FiaasV1() fiaasv1.FiaasV1Interface {
+	return c.fiaasV1
+}
+
+// Deprecated: Fiaas retrieves the default version of FiaasClient.
```
Comment (mortenlj) [inline pkg/client/clientset/versioned/clientset.go:47]: It seems strange to generate deprecated functions in a new library. Is this an artifact of the codegenerator, or can we disable it?

**Pull request merged**